decidim/decidim
# Add geocoding to budgets projects

#### :tophat: What? Why?
Participatory budget are often made by cities, and a lot of projects have a specific location in the city. The idea here is to allow administrators to add an address on projects so they can me displayed in a map on Budget component. This helps users to easily access projects of their neighbourhood.

Describe the solution you'd like

As an admin, I'd like to be able to enable geocoding on the Budget component.

Add a config flag in Budgets component configuration that allows the admin to enable/disable projects geocoding.
As an admin, I'd like to be able to add an address on the project I'm creating or editing.

Add an address field with geocoder in project creation & edition page. ([https://imgur.com/Dt2j5aR (External link)](https://imgur.com/Dt2j5aR))
As an admin, I'd like to be able to see the map preview on projects index in BO.

Add a "Map" column to display the map preview on projects index in BO (https://imgur.com/MdGtTBq (External link))
As a user, I'd like to see a map displaying geocoded projects on the projects index.

Show a map on projects index to display geocoded projects (ex here : https://imgur.com/34jVIGo (External link))
As a user, I'd like to see the address and the map preview on the project page.

As for meetings and proposals, add a card on project page to display address infos and map preview

#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #9109

#### Testing
* Create a project with an address
* Activate geocoding
* See maps with pin points

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](https://github.com/decidim/decidim/blob/develop/CONTRIBUTING.adoc) to this repository.


### :camera: Screenshots
Configuration in Budgets configuration
<img width="1068" alt="Capture d’écran 2022-05-10 à 16 23 09" src="https://user-images.githubusercontent.com/20232956/168726310-06b96e1e-dc2f-47a7-9b34-1e463c5ed943.png">

When geocoding is enabled and there is a project with an address, a pin is displayed on the map
<img width="1236" alt="Capture d’écran 2022-05-11 à 09 37 30" src="https://user-images.githubusercontent.com/20232956/168726337-de6af08b-438e-4cca-86a8-1f2921b8baa4.png">

When geocoding is enabled,a map is displayed on the projects index
<img width="1200" alt="Capture d’écran 2022-05-11 à 09 40 23" src="https://user-images.githubusercontent.com/20232956/168726468-c20ec0bb-35e1-4074-958b-cfad0cd73c81.png">

As for proposals, the card with address is displayed
<img width="1207" alt="Capture d’écran 2022-05-11 à 09 43 46" src="https://user-images.githubusercontent.com/20232956/168726490-199e25f2-c89f-48c1-b6b9-04dec95cda68.png">


:hearts: Thank you!
